### ISSUE TITLE
[question] identify user on route did transition

### ISSUE THREAD

**Issue by sescobb27** [josemarluedke/ember-cli-segment#28]
Hi, I have a question regarding calling `identifyUser` on every route `didTransition` action.
Is this something necessary? or is a workaround to any known issue or something? I'm asking because [here](https://segment.com/docs/spec/identify/#identities) they say `The identify call specifies a customer identity that you can reference across the customer’s whole lifetime.`, so, I don't see the need to call it every time.
Thanks.

**Comment by maletor** [comment]
I'm wondering this too. There is `aliasUser` though and the ability to disable via `defaultIdentifyUser`.

**Comment by sescobb27** [comment]
@maletor yes, I'm using that option right now and identifying our customers manually on login/signup, but it would be better if `identifyUser` had some kind of idempotence so it does not create so much noise in the analytics dashboards

**Comment by josemarluedke** [comment]
Interesting... I think we can come up with a good idea and replace that.

I'm open for suggestions and PRs are welcome.

**Comment by josemarluedke** [comment]
Maybe we can remove the automatic `identifyUser` and add documentation to manually call that in the sign in/sign up?

**Comment by sescobb27** [comment]
yes, I agree with removing automatic `identifyUser` and add documentation on how to do it manually, also we can tell the users to use `reset` on log out

**Comment by sbrichards** [comment]
I recommend keeping `identifyUser` on every route as the default behavior. I've personally experienced edge cases where that's hurt our user tracking.

Segment explains why they recommend that approach:
```

**Comment by denzo** [comment]
I do not think that is necessary. Their reasoning is not for a single page site like Ember.

> I log into your app. identifyis called. For whatever reason, I close the browser and don’t return until later. There’s no way of knowing where I will reenter your app from. I could start my session from anywhere.

In Ember no matter where you reenter the app the ApplicationRoute has to resolve. Thus calling `identifyUser` once is enough, assuming `currentUser` is available. If not, once the user loggedIn calling `identifyUser` after that should be enough.

**Comment by denzo** [comment]
@josemarluedke any thoughts on the above?

**Comment by raido** [comment]
@denzo I think it should be fine to call identifyUser just once.
